marc.go

```diff
@@ -313,3 +317,49 @@ func contentType(x byte) string {
 	}
 	return t
 }
+
+// RetrieveLanguageCodelist retrieves language codes for parsing MARC languages
+func RetrieveLanguageCodelist() (map[string]string, error) {
+	file, err := os.Open("fixtures/languages.xml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Language struct
+	type Language struct {
+		Name string `xml:"name"`
+		Code string `xml:"code"`
+	}
+
+	decoder := xml.NewDecoder(file)
+	languages := make(map[string]string)
+
+	for {
+		t, _ := decoder.Token()
+		if t == nil {
+			break
+		}
+		switch se := t.(type) {
+		case xml.StartElement:
+			if se.Name.Local == "language" {
+				var l Language
+				decoder.DecodeElement(&l, &se)
+				languages[l.Code] = l.Name
+			}
+		}
+	}
+	return languages, err
+}
+
+// TranslateLanguageCodes takes an array of MARC language codes and returns the language names.
+func TranslateLanguageCodes(x []string, languageCodes map[string]string) []string {
+	var languages []string
+	for _, l := range x {
+		name := languageCodes[l]
+		if name != "" {
+			languages = append(languages, name)
+		} else {
```

Comment:
I think this `else` block just inserts blank languages and doesn't seem to be useful. Maybe?

Reply:
Hmmm. In theory it should sometimes lead to adding the code I guess, but in practice it I can't find it doing that. With that in mind I'd leave it if we think the code should be kept if we can't find a lookup, but I think maybe a better thing to do would be to log a failed lookup so we could have the metadata team check for problems in the MARC.

To allow a passthrough of the code though and still not create empty languages we could do:
```
if l == "" {
  continue
}
```

at the start of this for loop.

Reply:
Unfortunately sometimes there are non-code strings in this field (example: "Italian or Latin words; also printed as text, with English translations, on p. li-cv."), so we need to retain all of the non-empty values that aren't translated.

I think it's better to handle empty values earlier in the parsing process, so that only fields with values get added to this array in the first place.